src/app/manager/hospital-map/room-map/room-map.component.ts

```diff
@@ -114,6 +131,26 @@ private isValidInput(): boolean {
  click(svg:any){
  
   }
+
+  public searchEquipmentInRoom(){
+    if(this.searchEquipmentInput.name == undefined || this.searchEquipmentInput.name == "") this.searchEquipmentInput.name = "0";
+
+     this.equipmentService.searchEquipmentInRoom(this.searchEquipmentInput).subscribe(res =>{
+      this.searchedEquipment = res;
+      if(this.searchEquipmentInput.name == "0") this.searchEquipmentInput.name = "";
+     })
+  }
+
+  public searchRoomsByFloorContainigEquipment(){
+    if(this.searchFloorInput.name == undefined || this.searchFloorInput.name == "") this.searchFloorInput.name = "0";
+    this.route.params.subscribe((params: Params) => {
+       this.searchFloorInput.roomId = params['id']
+       this.equipmentService.searchRoomsByFloorContainigEquipment(this.searchFloorInput).subscribe(res =>{
+       this.searchedRooms = res;
```

Comment:
This is a risky method because you can enter infinite loop with a mistake. Move the logic for route param subscription in ngOnInit and do the logic for searching inside of this method.